exportplan/serializers.py

```
@@ -146,9 +146,25 @@ def create(self, validated_data):
     def update(self, instance, validated_data):
 
         actions = {}
+
         if validated_data.get('export_plan_actions'):
             actions = validated_data.pop('export_plan_actions')
 
+        # This will allow partial updating to json fields during a patch update. Json fields generally represent
+        # a export plan page. we only want to update the field being sent else by nature we would wipe all the
+        # other fields
+        for field_name in validated_data.keys():
+            field_value = getattr(instance, field_name)
+            # Check if the field we are updating is JSON Type and ensure contents are JSON
+            if (
+                    isinstance(models.CompanyExportPlan._meta.get_field(field_name), JSONField) and
+                    isinstance(field_value, dict)
+            ):
```

don't you think we should raise the validation error if the above condition is not satisfied? also, worth adding negative test for this condition.

---

This isn't field validation, it's more a conditional statement 
1) source is JSON
2) Incoming is JOSN 
if these conditions are met we do partial update based on incoming keys. 
Else we do a normal field update. 

This is tested by an additional test by updating non Json fields.

---

Ahh, I see.